presentation/src/main/java/com/anytypeio/anytype/presentation/relations/CoverTypeMapper.kt

```diff
@@ -51,6 +52,54 @@ fun CoverWrapper.getCover(
     )
 }
 
+fun ObjectWrapper.Basic.cover(
+    urlBuilder: UrlBuilder,
+    coverImageHashProvider: CoverImageHashProvider
+): CoverView? {
+
+    val type = coverType
+
+    var coverColor: CoverColor? = null
+    var coverImage: Url? = null
+    var coverGradient: String? = null
+
+    when (type) {
+        CoverType.UPLOADED_IMAGE,
+        CoverType.UNSPLASH_IMAGE -> {
+            val targetObjectId = coverId
+            coverImage = if (!targetObjectId.isNullOrBlank()) {
+                urlBuilder.image(targetObjectId)
+            } else {
+                null
+            }
+        }
+        CoverType.BUNDLED_IMAGE -> {
+            val hash = coverId?.let { id ->
+                coverImageHashProvider.provide(id)
+            }
+            if (!hash.isNullOrBlank()) coverImage = urlBuilder.image(hash)
+        }
+        CoverType.COLOR -> {
+            coverColor = coverId?.let { id ->
+                CoverColor.values().find { it.code == id }
+            }
+        }
+        CoverType.GRADIENT -> {
+            coverGradient = coverId
+        }
+        CoverType.NONE -> {
+            // Do nothing.
```

Comment:
please pay attention to this function (private suspend fun getCoverContainer(), in case of None, the cover can be taken from image object or from relations

Reply:
will check this, thanks